main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -83,6 +84,31 @@ object TypeError {
     }
   }
 
+  /**
+   * Java ambiguous method type error.
+   *
+   * @param tpe0    the type of the receiver object.
+   * @param tpes    the types of the arguments.
+   * @param methods a list of possible candidate methods on the type of the receiver object.
+   * @param renv    the rigidity environment.
+   * @param loc     the location where the error occured.
+   */
+  case class AmbiguousMethod(methodName: String, tpe0: Type, tpes: List[Type], methods: List[Method], renv: RigidityEnv, loc: SourceLocation)(implicit flix: Flix) extends TypeError {
+    // TODO INTEROP better comment with candidate methods formatting
+    def summary: String = s"Ambiguous Java method '$methodName' in type '$tpe0' with arguments types (${tpes.mkString(", ")})."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Java method '$methodName' from type '${red(formatType(tpe0, Some(renv)))}' with arguments types (${tpes.mkString(", ")}) is ambiguous.
+         | Possible candidate methods:
+         |  ${methods.map(m => s"${m.getName}(${m.getParameterTypes.map(t => t.getName).mkString(", ")})").mkString(", ")}
```

Comment:
This is a lot to do inline. Can we move this into a helper?

Reply:
I’ll do this, thank you. Do I do it in as a nested function or outside but in the same file?

Reply:
A local function is probably fine. We can move it if we need it elsewhere.